### DISCUSSION TITLE
DNS Checking at Regular Periods

### DISCUSSION THREAD

**Discussion by TR-SametOzen** [postalserver/postal#3157] [Development discussions]
Hello,

How can I make my Postal Server check DNS at regular intervals?

![image](https://github.com/user-attachments/assets/e7aee2a4-1aab-4616-afa6-2b125b610f8d)

**Comment by willpower232** [comment]
I think it does that by default anyway, what problem are you trying to solve?

**Reply by TR-SametOzen** [reply]
Hello,

It does not check unless the button in the picture is clicked.

I want it to do this automatically.

![image](https://github.com/user-attachments/assets/59bc6c1f-436a-4aa9-9d54-feefcd139f9e)

**Reply by willpower232** [reply]
If you have checked it at least once yourself by clicking that button then it should be happening automatically every hour

https://github.com/postalserver/postal/blob/main/app/scheduled_tasks/check_all_dns_scheduled_task.rb

**Reply by TR-SametOzen** [reply]
Hello,

Yes, but I want the system to automatically scan every domain I add without clicking this button.

So, is there a chance to add a domain here via API?

**Reply by willpower232** [reply]
There isn't an API for this currently so you would need to add the domains directly in the web UI in order to get the right DKIM record.

**Reply by TR-SametOzen** [reply]
So how can I enable continuous control without clicking the button?
